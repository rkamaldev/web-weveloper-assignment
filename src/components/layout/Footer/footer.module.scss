.footerComponent {
  background-color: #dae2f4;
  width: 100%;
  padding: 10px;
  min-height: 60px;
  position: absolute;
  bottom: 0;
  display: flex;
  align-items: stretch;

  div {
    min-height: 40px;
    height: 40px;
    padding: 10px 10px;
    border-right: solid 1px #ccc;
    font-family: monospace;

    .link {
      color: #23366f;
      text-decoration: none;
      font-weight: bold;
    }

    .link:hover {
      color: #5495cf;
    }

    .text {
    }
  }

  div:last-child {
    border-right: none;
  }
}
