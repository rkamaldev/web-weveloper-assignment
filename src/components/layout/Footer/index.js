import React from 'react';
import Link from 'next/link';
import PropTypes from 'prop-types';

import styles from './footer.module.scss';

function Footer(props) {
  const { footerItems } = props;

  return (
    <div className={styles.footerComponent}>
      {footerItems &&
        footerItems.length &&
        footerItems.map((item, index) => (
          <div key={index}>
            {item.type === 'link' && (
              <Link className={styles.link} href={item.link} target={item.target}>
                {item.content}
              </Link>
            )}

            {item.type === 'text' && <span className={styles.text}>{item.content}</span>}
          </div>
        ))}
    </div>
  );
}

Footer.propTypes = {
  footerItems: PropTypes.any,
};

export default Footer;
