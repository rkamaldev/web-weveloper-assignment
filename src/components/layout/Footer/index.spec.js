/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';

import Footer from './';

describe('#Footer', () => {
  it('Should render Footer component with given config', () => {
    const component = renderer.create(
      <Footer
        footerItems={[
          {
            type: 'link',
            link: 'https://www.zurich.com/contact-us',
            content: 'Contact us',
            target: '',
          },
          {
            type: 'link',
            link: 'https://www.zurich.com/services/privacy',
            content: 'Privacy',
            target: '_blank',
          },
          {
            type: 'text',
            content: 'Â© 2023 Rights Reserved',
          },
        ]}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render Footer component without config', () => {
    const component = renderer.create(<Footer />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
