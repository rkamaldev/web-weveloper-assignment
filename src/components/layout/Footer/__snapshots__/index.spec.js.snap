// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Footer Should render Footer component with given config 1`] = `
<div
  className="footerComponent"
>
  <div>
    <a
      className="link"
      href="https://www.zurich.com/contact-us"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target=""
    >
      Contact us
    </a>
  </div>
  <div>
    <a
      className="link"
      href="https://www.zurich.com/services/privacy"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      Privacy
    </a>
  </div>
  <div>
    <span
      className="text"
    >
      © 2023 Rights Reserved
    </span>
  </div>
</div>
`;

exports[`#Footer Should render Footer component without config 1`] = `
<div
  className="footerComponent"
/>
`;

exports[`Should render Footer component with given config 1`] = `
<div
  className="footerComponent"
>
  <div>
    <a
      className="link"
      href="https://www.zurich.com/contact-us"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target=""
    >
      Contact us
    </a>
  </div>
  <div>
    <a
      className="link"
      href="https://www.zurich.com/services/privacy"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      target="_blank"
    >
      Privacy
    </a>
  </div>
  <div>
    <span
      className="text"
    >
      © 2023 Rights Reserved
    </span>
  </div>
</div>
`;
